Python/Product/PythonTools/PythonTools/Intellisense/CompletionComparer.cs

```diff
@@ -96,17 +93,33 @@ class CompletionComparer : IEqualityComparer<CompletionResult>, IComparer<Comple
         }
 
         /// <summary>
-        /// Compares two <see cref="MemberResult"/> structures for equality.
+        /// Compares two <see cref="CompletionResult"/> structures for equality.
         /// </summary>
         public bool Equals(CompletionResult x, CompletionResult y) {
             return x.Name.Equals(y.Name);
         }
 
         /// <summary>
-        /// Gets the hash code for a <see cref="MemberResult"/> structure.
+        /// Gets the hash code for a <see cref="CompletionResult"/> structure.
         /// </summary>
         public int GetHashCode(CompletionResult obj) {
             return obj.Name.GetHashCode();
         }
     }
+
+    /// <summary>
+    /// A comparer for use with <see cref="Enumerable.Union{TSource}(IEnumerable{TSource}, IEnumerable{TSource}, IEqualityComparer{TSource})"/>
+    /// to eliminate duplicate items with the same <see cref="CompletionResult.MergeKey"/>.
+    /// </summary>
+    class CompletionMergeKeyComparer : IEqualityComparer<CompletionResult> {
+        public static readonly CompletionMergeKeyComparer Instance = new CompletionMergeKeyComparer();
+
+        public bool Equals(CompletionResult x, CompletionResult y) {
+            return x.MergeKey.Equals(y.MergeKey);
```

Comment:
Is it possible that `x.MergeKey == null`?

Reply:
No. I'll add a check for null in CompletionResult ctor.